support-services/src/main/scala/com/gu/support/paperround/PaperRoundService.scala

```diff
@@ -0,0 +1,193 @@
+package com.gu.support.paperround
+
+import com.gu.okhttp.RequestRunners.FutureHttpClient
+import com.gu.rest.WebServiceHelper
+import com.gu.support.config.PaperRoundConfig
+import java.time.ZonedDateTime
+import io.circe._
+import io.circe.generic.semiauto._
+import io.circe.generic.extras.semiauto._
+import io.circe.generic.extras.decoding.ConfiguredDecoder
+import io.circe.generic.extras.Configuration
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import PaperRoundService.{AgentsEndpoint, CoverageEndpoint, ChargeBandsEndpoint}
+
+class PaperRoundService(config: PaperRoundConfig, client: FutureHttpClient)(implicit
+    executionContext: ExecutionContext,
+) extends WebServiceHelper[PaperRoundService.Error] {
+  override val wsUrl: String = config.apiUrl
+  override val httpClient: FutureHttpClient = client
+
+  def coverage(body: CoverageEndpoint.RequestBody): Future[CoverageEndpoint.Response] = {
+    postForm[CoverageEndpoint.Response](
+      endpoint = "coverage",
+      data = Map("postcode" -> List(body.postcode)),
+      headers = Map("x-api-key" -> config.apiKey),
+    )
+  }
+
+  def agents(): Future[AgentsEndpoint.Response] = {
+    postForm[AgentsEndpoint.Response](
+      endpoint = "agents",
+      data = Map.empty,
+      headers = Map("x-api-key" -> config.apiKey),
+    )
+  }
+
+  def chargebands(): Future[ChargeBandsEndpoint.Response] = {
+    postForm[ChargeBandsEndpoint.Response](
+      endpoint = "chargebands",
+      data = Map.empty,
+      headers = Map("x-api-key" -> config.apiKey),
+    )
+  }
+}
+
+object PaperRoundService {
+  val snakeCase: Configuration = Configuration.default.withSnakeCaseMemberNames
+  val lowercase: Configuration = Configuration.default.copy(transformMemberNames = _.toLowerCase)
+  case class Error(statusCode: Integer, message: String, errorCode: ZonedDateTime)
```

Comment:
Are we planning to ask PaperRound why is there the ZonedDateTime  passed as an errorCode?

Reply:
Yes, I’ll email Chris today